src/Makefile

```
@@ -27,5 +27,8 @@ test: $(patsubst %.c, .run-%.c, $(wildcard *_test.c))
 %_test: %_test.c minunit.h
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
+frontend.out: frontend.c frontend.h
+	$(CC) frontend.c $(CFLAGS) $(LDFLAGS) -o frontend.out
+
```

> - We should talk about how to setup the makefile for the frontend, and other things that might come up. I did it naively, not sure if we want to be more fancy.

I'd recommend something like this, which won't link other executables to `ncurses` and means we don't have to write the same build command for everything:
```suggestion
frontend: LDFLAGS+=-lncurses
```
That way we can use [Make's builtin rules](https://www.gnu.org/software/make/manual/html_node/Catalogue-of-Rules.html#Catalogue-of-Rules), the one's for C look like this:
```Makefile
%: %.c
	$(LINK.c) $^ $(LOADLIBES) $(LDLIBS) -o $@

LINK.c = $(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(TARGET_ARCH)
```
One thing I just learned is that [`LDFLAGS` is for configuration flags, while `LDLIBS` is for libraries to load](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html), so we should also fix line 4.

---

The `LDFLAGS` vs `LDLIBS` thing came back to bite me for homework 08:

This doesn't work:
```Makefile
ex_compile: CFLAGS+=$(shell pkg-config --cflags glib-2.0)
ex_compile: LDFLAGS+=$(shell pkg-config --libs glib-2.0)
```
But this does:
```Makefile
ex_compile: CFLAGS+=$(shell pkg-config --cflags glib-2.0)
ex_compile: LDLIBS+=$(shell pkg-config --libs glib-2.0)
```